SConstruct

```diff
@@ -1470,12 +1470,21 @@ if libraryPathEnvVar :
 	testEnv["ENV"][libraryPathEnvVar] = testEnvLibPath
 testEnv["ENV"]["IECORE_OP_PATHS"] = "test/IECore/ops"
 
-testEnv.Append(
-	LIBS = [
-		"boost_unit_test_framework${BOOST_LIB_SUFFIX}",
-		"boost_test_exec_monitor${BOOST_LIB_SUFFIX}"
-	],
-)
+c = Configure( testEnv )
+
+withBoostUnitTest = c.CheckLibWithHeader( env.subst( "boost_unit_test_framework" + env["BOOST_LIB_SUFFIX"] ), "boost/test/unit_test.hpp", "CXX" )
+withBoostTestExecMonitor = c.CheckLibWithHeader( env.subst( "boost_test_exec_monitor" + env["BOOST_LIB_SUFFIX"] ), "boost/test/test_exec_monitor.hpp", "CXX" )
+withBoostTest = withBoostUnitTest and withBoostTestExecMonitor
+
+c.Finish()
+
+if withBoostTest:
+	testEnv.Append(
+		LIBS = [
+			"boost_unit_test_framework${BOOST_LIB_SUFFIX}",
+			"boost_test_exec_monitor${BOOST_LIB_SUFFIX}"
+		],
```

Comment:
I was under the impression that `CheckLibWithHeader()` automatically added the libs, in which case this would be redundant.

Reply:
I tested without those lines and you are correct, they are in fact redundant. I've removed them now, and I've removed a similar redundant append for libboost_wave too.